fuel-vm/src/gas.rs

```diff
@@ -124,3 +136,361 @@ impl GasUnit {
         Self::Accumulated(self.cost() + other.cost())
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Gas costings for every op.
+/// The inner values are wrapped in an [`Arc`]
```

Comment:
What are the performance implications of checking gas costs though through an `Arc`, since while it will be cheap to clone I'd imagine this is quite a bit more expensive than referencing constant values for Gas costs

Reply:
The Arc is probably insignificant, but we could add a todo here to optimize this using lifetimes or a regular `Rc` since there's no concurrency in the VM.

Reply:
Can we make a perf issue to then move these to `lazy_static` once things stabilize then? I understand why it is like this for now, but as we push more towards the efficiency frontier it may be helpful to remember this

Reply:
lazy_static alone may be insufficient, would require some investigation to find the exact solution.

Reply:
#302

Reply:
`Deref` of an `Arc` is just de-refencing a pointer so it won't be any quicker too do a static or `Rc`

Reply:
Just FYI there's no way to try and get const performance when we are passing in values at runtime.

Reply:
> Just FYI there's no way to try and get const performance when we are passing in values at runtime.

*without unsafe

Reply:
I think you are confusing static and const. Const has to be known at compile time as it's in-lined. You can make a static global for the gas costs but it will still require a pointer de-reference so it will not be any faster then an Arc.

Reply:
ah your right, i was thinking of static mut